components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSOAPUtils.java

```
@@ -0,0 +1,281 @@
+            Element element = document.getDocumentElement();
+            UnmarshallerFactory unmarshallerFactory = Configuration.getUnmarshallerFactory();
+            Unmarshaller unmarshaller = unmarshallerFactory.getUnmarshaller(element);
+            return unmarshaller.unmarshall(element);
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            String err = "Error in Processing the  SAML request";
+            throw new IdentitySAML2ECPException(err);
+        }
+    }
+
+    /**
+     *
+     * Decode the request recived by the /ecp servlet.
+     * Validate the SOAP message
+     * Check whether the SOAP body contains a valid SAML request
+     * @param soapMessage
+     * @return
+     * @throws IdentitySAML2ECPException
+     */
+    public static String decodeSOAPMessage(SOAPMessage soapMessage) throws IdentitySAML2ECPException {
+        SOAPBody body = null;
+        String samlRequest = null;
+        String message = null;
+        try {
+            body = soapMessage.getSOAPPart().getEnvelope().getBody();
+        } catch (SOAPException e) {
+            log.error("Error Processing the SOAP Mesage");
+            throw new IdentitySAML2ECPException(e.getMessage());
+        }
+
+        int elementSize = 0;
+        try {
+            Iterator<?> elements = body.getChildElements();
+            while (elements.hasNext()) {
+                SOAPElement element = (SOAPElement) elements.next();
+                DOMSource source = new DOMSource(element);
+                StringWriter stringResult = new StringWriter();
+                try {
+                    TransformerFactory.newInstance().newTransformer().transform(source, new StreamResult(stringResult));
+                } catch (TransformerException e) {
+                    String err = "Transformer Exception";
+                    log.error(err);
+                    throw new IdentitySAML2ECPException(e.getMessage());
```

As a general comment : We should not swallow the original exception. We should either log it, or wrap and re-throw it. Let's fix this in overall PR.

---

> As a general comment : We should not swallow the original exception. We should either log it, or wrap and re-throw it. Let's fix this in overall PR.

fixed